client/src/app.tsx

```diff
@@ -108,9 +109,15 @@ function PageOrPageNotFound({ pageNotFound, children }) {
   );
 }
 
-export function App(appProps) {
+export function App(appProps: HydrationData) {
+  const { pathname } = useLocation();
+  const initialPathname = React.useRef(pathname);
+  const pageNotFound =
+    (appProps.pageNotFound || false) && initialPathname.current === pathname;
```

Comment:
This means for client-side navigation, `pageNotFound` will always be `false`?

Reply:
Yes